server/test/org/apache/cloudstack/network/lb/CertServiceTest.java

```diff
@@ -348,90 +348,91 @@ public void runUploadSslCertNoRootCert() throws IOException, IllegalAccessExcept
+        final Account account = new AccountVO("testaccount", 1, "networkdomain", (short)0, UUID.randomUUID().toString());
         when(certService._accountMgr.getAccount(anyLong())).thenReturn(account);
 
         certService._domainDao = Mockito.mock(DomainDao.class);
-        DomainVO domain = new DomainVO("networkdomain", 1L, 1L, "networkdomain");
+        final DomainVO domain = new DomainVO("networkdomain", 1L, 1L, "networkdomain");
         when(certService._domainDao.findByIdIncludingRemoved(anyLong())).thenReturn(domain);
 
         certService._sslCertDao = Mockito.mock(SslCertDao.class);
         when(certService._sslCertDao.persist(any(SslCertVO.class))).thenReturn(new SslCertVO());
 
         //creating the command
-        UploadSslCertCmd uploadCmd = new UploadSslCertCmdExtn();
-        Class<?> _class = uploadCmd.getClass().getSuperclass();
+        final UploadSslCertCmd uploadCmd = new UploadSslCertCmdExtn();
+        final Class<?> klazz = uploadCmd.getClass().getSuperclass();
 
-        Field certField = _class.getDeclaredField("cert");
+        final Field certField = klazz.getDeclaredField("cert");
         certField.setAccessible(true);
         certField.set(uploadCmd, cert);
 
-        Field keyField = _class.getDeclaredField("key");
+        final Field keyField = klazz.getDeclaredField("key");
         keyField.setAccessible(true);
         keyField.set(uploadCmd, key);
 
-        Field passField = _class.getDeclaredField("password");
+        final Field passField = klazz.getDeclaredField("password");
         passField.setAccessible(true);
         passField.set(uploadCmd, password);
 
         try {
             certService.uploadSslCert(uploadCmd);
             fail("Given an encrypted private key with a bad password. Upload should fail.");
-        } catch (Exception e) {
-            assertTrue(e.getMessage().contains("please check password and data"));
+        } catch (final Exception e) {
+            assertTrue("Did not expect message: " + e.getMessage(),
+                    e.getMessage().contains("Error parsing certificate data Parsing certificate/key failed: Invalid Key format."));
         }
 
     }
 
     @Test
     public void runUploadSslCertBadkeyPair() throws IOException, IllegalAccessException, NoSuchFieldException {
         // Reading appropritate files
-        String certFile = URLDecoder.decode(getClass().getResource("/certs/rsa_self_signed.crt").getFile(),Charset.defaultCharset().name());
-        String keyFile = URLDecoder.decode(getClass().getResource("/certs/non_root.key").getFile(),Charset.defaultCharset().name());
+        final String certFile = URLDecoder.decode(getClass().getResource("/certs/rsa_self_signed.crt").getFile(),Charset.defaultCharset().name());
+        final String keyFile = URLDecoder.decode(getClass().getResource("/certs/non_root.key").getFile(),Charset.defaultCharset().name());
```

Comment:
Please see line notes for 154-155 as those comments apply to lines 401-402 as well.